File: functions/js/matrix/matrix.js
function multiplyMatrices(m1, m2) {
    var result = [];
    for (var i = 0; i < m1.length; i++) {
        result[i] = [];
        for (var j = 0; j < m2[0].length; j++) {
            var sum = 0;
            for (var k = 0; k < m1[0].length; k++) {
                sum += m1[i][k] * m2[k][j];
            }
            result[i][j] = sum;
        }
    }
    return result;
}

function generateMatrix(size) {
    var matrix = [];
    for (var i = 0; i < size; i++) {
        var row = [];
        for (var j = 0; j < size; j++) {
            row.push(Math.round(117 + Math.random() * 40));
        }
        matrix.push(row);
    }
    return matrix;
}

function main(params) {
    if (!params.size) {
        return {
            result: "Matrix size is required"
        }
    }
    if (!params.ramLoader) {
        return {
            result: "ramLoader is required"
        }
    }

    var fakeList = [];

    var m1 = generateMatrix(params.size);
    var m2 = generateMatrix(params.size);

    for (var k = 0; k < params.ramLoader; k++) {
        fakeList.push(m1[k % 9].slice(0))
    }

    var result = multiplyMatrices(m1, m2);

    return {
    	result: 'Success'
    };
}

File: functions/java/Matrix.java
import com.google.gson.JsonObject;
import java.util.ArrayList;
import java.util.List;

public class Matrix {
    private static long[][] multiplyMatrices(long[][] m1, long[][] m2, int size) {
        long[][] result = new long[size][size];
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                int sum = 0;
                for (int k = 0; k < size; k++) {
                    sum += m1[i][k] * m2[k][j];
                }
                result[i][j] = sum;
            }
        }
        return result;
    }

    private static long[][] generateMatrix(int size) {
        long[][] matrix = new long[size][size];
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                long number = Math.round(117 + Math.random() * 40);
                matrix[i][j] = number;
            }
        }
        return matrix;
    }

    public static JsonObject main(JsonObject args) {
        JsonObject response = new JsonObject();
        if (args.has("size") && args.has("ramLoader")) {
            int size = args.getAsJsonPrimitive("size").getAsInt();
            long ramLoader = args.getAsJsonPrimitive("size").getAsLong();

            long[][] m1 = generateMatrix(size);
            long[][] m2 = generateMatrix(size);

            List<Object> fakeList = new ArrayList<>();

            for (int k = 0; k < ramLoader; k++) {
                fakeList.add(m1[k % 9].clone());
            }

            multiplyMatrices(m1, m2, size);
            response.addProperty("result", "Success");
        } else {
            response.addProperty("result", "No input number was provided");
        }
        return response;
    }
}
